authorization/v2/uniformResourceIdentifier_test.go

```diff
@@ -42,3 +42,153 @@ func TestNewUriFromRequest(t *testing.T) {
 	}, uri.arguments)
 	assert.Nil(t, uri.metadata)
 }
+
```

Comment:
Could you add unit tests for more tricky situations?

E.g.: `hs:hs_auth:api:v2:provide_access_to_uri?`, `hs:hs_auth:api:v2:provide_access_to_uri?test=ok%23#test2=ok` (encoded `#` inside arguments),  `hs:hs_auth:api:v2:provide_access_to_uri?test=`

Reply:
> hs:hs_auth:api:v2:provide_access_to_uri?test=

Should this be a valid URI? Would this indicate the key `test` has a value of `""`?

Reply:
Yes, that's fine. As far as I can tell, it's the only way to represent an empty string, right?

Reply:
Yep, I believe so